source/base/function_spherical.cc

```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```

Comment:
But you added the `component` argument to the `svalue()`, `sgradient()`, and `shessian()` functions. Why not just pass the component through to these functions? Right now, these arguments are all unused. (I see now that the `value()` and `gradient()` functions also ignore the component...)